<?xml version="1.0"?>

<!-- included by, and macro expanded in, diffbot_urdf.xacro -->

<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="diffbot_ros2_control" params="name">

    <ros2_control name="${name}" type="system">

      <hardware>
        <plugin>diffbot/DiffbotSystem</plugin>
        <!-- used by both wheels -->
        <param name="ser_dev_name">/dev/ttyAMA0</param>
        <param name="steps_per_rev">3200</param>
      </hardware>

      <joint name="left_wheel_joint">
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="right_wheel_joint">
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

    </ros2_control>

  </xacro:macro>

</robot>
